### DISCUSSION TITLE
使用 Docker Compose 部署 LobeChat 服务端数据库版本时候遇到Logto 404的错误

### DISCUSSION THREAD

**Discussion by vannchic** [lobehub/lobehub#3678] [Self Hosting] [answered]
### 📦 Environment

Docker

### 📌 Version

v1.14.9

### 💻 Operating System

Other Linux

### 🌐 Browser

Chrome

### 🐛 Bug Description

访问: `http://lobe-auth-ui.example.com/` 时候 302到 `http://lobe-auth-api.example.com/unknown-session`
然后又报错404了
如图:
![image](https://github.com/user-attachments/assets/13bcad57-0e01-4fde-bce6-54f411c5e315)

docker compose使用的是:
```
curl -fsSL https://raw.githubusercontent.com/lobehub/lobe-chat/HEAD/docker-compose/production/docker-compose.yml > docker-compose.yml
curl -fsSL https://raw.githubusercontent.com/lobehub/lobe-chat/HEAD/docker-compose/production/.env.zh-CN.example > .env
```
相关域名默认用的是demo的域名,仅仅做了如下调整:
- 去掉了HTTPS
- 做了反向代理

应该不是反向代理的问题,因为minio的 ui界面就可以正常访问
http://lobe-s3-ui.example.com/login

docker ps情况如下:

![image](https://github.com/user-attachments/assets/4df8c31b-47b7-4c53-8bfa-e22112cc4346)

### 📷 Recurrence Steps

- git clone
- 下载docker-composer
```
curl -fsSL https://raw.githubusercontent.com/lobehub/lobe-chat/HEAD/docker-compose/local/docker-compose.yml > docker-compose.yml
curl -fsSL https://raw.githubusercontent.com/lobehub/lobe-chat/HEAD/docker-compose/local/.env.zh-CN.example > .env
``` 
- 修改docker-compose 和 .env 里的 https 都调整为 http
- docker-composer up
- 配置反向代理
- 访问:lobe-auth-ui.example.com　跳转到　http://lobe-auth-api.example.com/unknown-session　并且　４０４

### 🚦 Expected Behavior

可以正常打开Logto

[a comment by an automated account is not shown]

**Comment by SpeedupMaster** [comment]
域名解析在Cloudflare上？

[a comment by an automated account is not shown]

**Comment by vannchic** [comment]
> 域名解析在Cloudflare上？

不是的, 本机配置的域名. window下使用vm安装的centos系统,docker跑在centos上的. ip是192.168.1.129
192.168.1.129 lobe.example.com lobe-chat.test lobe-auth-api.example.com lobe-auth-ui.example.com lobe-s3-api.example.com lobe-s3-ui.example.com

[a comment by an automated account is not shown]

**Comment by SpeedupMaster** [comment]
windows上不是可以直接本地部署吗？没必要在虚拟机上部署吧。

[a comment by an automated account is not shown]

**Comment by vannchic** [comment]
> windows上不是可以直接本地部署吗？没必要在虚拟机上部署吧。

写后端比较多一些.之前一直在centos下开发.  

不过这个环境理论上应该也是能支持的吧[实际我们到时候内网部署时候,也是类似的情况,内网的centos,内网的域名]?  

不是很理解为啥会404

docker logs 里能看到Logto 是输出成功的,没有报错

[a comment by an automated account is not shown]

**Comment by SpeedupMaster** [comment]
你这个问题我也遇到过，我是域名解析在CF上，开启了小黄云代理。在CF上把代理关了就没问题了。
CF上设置重写规则也可以解决，你可以参考下
![image](https://github.com/user-attachments/assets/4af91a53-1237-4109-9c51-8a53a4927ef8)

[a comment by an automated account is not shown]

**Comment by JonFanne** [comment]
这个我也遇到了，3001和3002全部反代理就解决了

[a comment by an automated account is not shown]

**Comment by vannchic** [comment]
> 这个我也遇到了，3001和3002全部反代理就解决了

我上午都配置了的反向代理,依然还是 404. 
lobe-auth-ui.example.com 的
![image](https://github.com/user-attachments/assets/588d9f96-857a-460a-841c-b4c8baa28c7c)

lobe-auth-api.example.com 的
![image](https://github.com/user-attachments/assets/7fb82d23-ab93-4eab-bed0-a765d00a3d49)

**Comment by JonFanne** [comment]
建个用户试一下

在 2024年8月30日星期五，vannchic ***@***.***> 写道：

**Comment by vannchic** [comment]
> 建个用户试一下 在 2024年8月30日星期五，vannchic ***@***.***> 写道：
> […](#)
> 这个我也遇到了，3001和3002全部反代理就解决了 我上午都配置了的反向代理,依然还是 404. lobe-auth-ui.example.com 的 image.png (view on web) <https://github.com/user-attachments/assets/588d9f96-857a-460a-841c-b4c8baa28c7c> lobe-auth-api.example.com 的 image.png (view on web) <https://github.com/user-attachments/assets/7fb82d23-ab93-4eab-bed0-a765d00a3d49> — Reply to this email directly, view it on GitHub <[#3673 (comment)](https://github.com/lobehub/lobe-chat/issues/3673#issuecomment-2320344699)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADYYUAWBSE4KNFQ6YH6UZGTZUAN33AVCNFSM6AAAAABNLS3D62VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMRQGM2DINRZHE> . You are receiving this because you commented.Message ID: ***@***.***>

没理解你说的建用户. 
访问 https://lobe-auth-ui.example.com 时候直接404了, 没法下一步了
![image](https://github.com/user-attachments/assets/9d4050ad-c346-4c0c-ba54-0ead233f1a82)

**Comment by qq940500529** [comment]
我也遇到同样的问题，看到你也使用了宝塔面板做的反向代理，现在我怀疑是宝塔面板造成的

**Comment by jtsang4** [comment, marked as answer]
如果是反代 https 的可以参考这个 issue，加个 Ngnix 配置就好了 https://github.com/logto-io/logto/issues/4279

**Reply by qq940500529** [reply]
这个方法可以，已经解决了👍

**Reply by jueinin** [reply]
光加header还不行，似乎端口也必须是443？

**Comment by NikolaCavendish** [comment]
如果是宝塔反代https可以参考这个问题[lobehub/lobe-chat/issues/6569](https://github.com/lobehub/lobe-chat/issues/6569)，Nginx文件注释一行就好了。
我今天也莫名其妙的反代出现404，直接访问端口没问题，反代后的地址有问题，把这个缓存关掉就好了。
